renderer/utils/reports.ts

```diff
@@ -216,3 +216,65 @@ export function validation(activities: Array<ReportActivity>) {
   }
   return activities;
 }
+
+export function addingSuggestions(
+  activities: Array<ReportActivity> | null,
+  latestProjAndDesc: Record<string, [string]>,
+  latestProjAndAct: Record<string, [string]>
+) {
+  for (let i = 0; i < activities.length; i++) {
+    if (!activities[i].project || activities[i].project?.startsWith("!")) {
+      continue;
+    }
+    const projectKey = activities[i].project.trim();
+
+    if (
+      Object.keys(latestProjAndDesc).length &&
+      !latestProjAndDesc.hasOwnProperty(projectKey) &&
+      activities[i].description &&
+      !latestProjAndAct.hasOwnProperty(projectKey)
```

Comment:
mb you can simplified all this "if" below. I mean you always have at least "Object.keys(latestProjAndDesc).length" and  "activities[i].description". Perform this check before all this "ifs"